src/pages/blocks/[blockId]/index.tsx

```diff
@@ -0,0 +1,211 @@
+import { Transaction } from '@defichain/whale-api-client/dist/api/transactions'
+
+import { Breadcrumb } from '@components/commons/Breadcrumb'
+import { OverflowTable } from '@components/commons/OverflowTable'
+import { CursorPage, CursorPagination } from '@components/commons/CursorPagination'
+import { getWhaleApiClient } from '@contexts/WhaleContext'
+import { IoCopyOutline } from 'react-icons/io5'
+
+import { fromUnixTime, format } from 'date-fns'
+
+interface BlockDetailsPageProps {
+  block: Block
+  confirmations: number
+  transactions: {
+    items: Transaction[]
+    pages: CursorPage[]
+  }
+}
+
+function CopyButton ({ text }: { text: string }): JSX.Element {
+  const [open, setOpen] = useState<Boolean>(false)
+
+  async function copy (): Promise<void> {
+    await navigator.clipboard.writeText(text)
+    setOpen(true)
+
+    setTimeout(() => {
+      setOpen(false)
+    }, 500)
+  }
+
+  return (
+    <div className='ml-1'>
+      <button className='cursor-pointer outline-none p-1 bg-gray-100 rounded shadow-sm' onClick={copy}>
+        <IoCopyOutline className='h-5 w-5 text-gray-500' />
+      </button>
+      {
+        open === true
+          ? (
+            <span className='bg-gray-100 p-1 rounded absolute mr-4'>
+              Copied!
+            </span>
+          )
+          : null
+      }
+    </div>
+  )
+}
+
+export default function BlockDetails ({ block, confirmations, transactions }: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  const leftBlockDetails = [
+    // {
+    //   label: 'Block Reward:',
+    //   value: 'xxxx DFI'
+    // },
+    {
+      label: 'Height:',
+      value: `${block.height}`,
+      testId: 'block-detail-height'
+
+    },
+    {
+      label: 'Transactions:',
+      value: `${block.transactionCount}`,
+      testId: 'block-detail-transactions'
+    },
+    {
+      label: 'Timestamp:',
+      value: `${format(fromUnixTime(block.medianTime), 'PPpp')}`,
+      testId: 'block-detail-timestamp'
+    },
+    {
+      label: 'Confirmations:',
+      value: `${confirmations}`,
+      testId: 'block-detail-confirmations'
+    },
+    {
+      label: 'Merkle Root:',
+      content: (
+        <>
+          <span className='flex-grow break-all' data-testid='block-detail-merkle-root'>{block.merkleroot}</span>
+          <CopyButton text={block.merkleroot} />
+        </>
+      )
+    }
+  ]
+
+  const rightBlockDetails = [
+    {
+      label: 'Difficulty:', value: `${block.difficulty}`, testId: 'block-detail-difficulty'
+    },
+    // {
+    //   label: 'Bits:', value: `${block.weight}`
+    // },
+    {
+      label: 'Size (bytes):', value: `${block.size}`, testId: 'block-detail-size'
+    },
+    {
+      label: 'Version:', value: `${block.version}`, testId: 'block-detail-version'
+    }
+    // {
+    //   label: 'Next Block:', value: `${block.nextBlock.height}`
+    // },
+    // {
+    //   label: 'Previous Block:', value: `${block.prevBlock.height}`
+    // }
+
+  ]
```

Comment:
Don't do this, Create a Component instead

for each Component, don't for each the HTML

Reply:
I realized my sentences can be misleading, in terms of `Component`, avoid creating shared global components when you don't have to. However, always opt to better organize your code via internal Component within the same file.

For `array.map()` avoid passing an array and returning `JSX.Element[]`. However, you should `function Component(props: {}): JSX.Element`. Functional component in React has special property, you should always opt to use it. For more information, take a look at this issue: https://github.com/DeFiCh/wallet/issues/694